pkg/kubelet/managed/managed.go

```diff
@@ -117,14 +117,29 @@ func GenerateResourceName(workloadName string) v1.ResourceName {
 func updateContainers(workloadName string, pod *v1.Pod) error {
 	updateContainer := func(container *v1.Container) error {
 		if container.Resources.Requests == nil {
-			return fmt.Errorf("managed container %v does not have Resource.Requests", container.Name)
+			// Nothing to modify, but that is OK, it will not
+			// change the QoS class of the modified Pod
+			return nil
 		}
-		if _, ok := container.Resources.Requests[v1.ResourceCPU]; !ok {
+
+		_, cpuOk := container.Resources.Requests[v1.ResourceCPU]
+		_, memoryOk := container.Resources.Requests[v1.ResourceMemory]
+
+		// When both cpu and memory requests are missing, there is nothing
+		// to do
+		if !cpuOk && !memoryOk {
```

Comment:
Isn't this missing the case where only limits are set and requests are implied from limits?

Reply:
But in such case it is a guaranteed QoS container, right? I do not think the workload partitioning was meant for those, because the requests -> annotation conversion that strips the cpu requests would make the Pod Burstable anyway.

And since the code here converts requests into annotations and strips the original cpu requests (orig line 148/149) it would convert the container further down in class to Best-effort if no memory is specified.

But you might be right that I missed a corner case where cpu has requests only and memory has limits only.